quinn-proto/src/connection/mod.rs

```diff
@@ -3127,7 +3219,16 @@ impl Connection {
                 trace!("PATH_CHALLENGE {:08x}", token);
                 buf.write(frame::FrameType::PATH_CHALLENGE);
                 buf.write(token);
-                self.stats.frame_tx.path_challenge += 1;
+
+                send_observed_address(
```

Comment:
How do we know this doesn't result in redundant frames being included in a single packet? Would it be easier to make the decision to include this frame in exactly one place?